pkgs/applications/version-management/git-and-tools/git/default.nix

```diff
@@ -141,8 +141,9 @@ stdenv.mkDerivation {
       cp -a contrib $out/share/git/
       mkdir -p $out/share/emacs/site-lisp
       ln -s "$out/share/git/contrib/emacs/"*.el $out/share/emacs/site-lisp/
+      mkdir -p $out/share/bash-completion/completions
+      ln -s $out/share/git/contrib/completion/git-completion.bash $out/share/bash-completion/completions/git
       mkdir -p $out/etc/bash_completion.d
-      ln -s $out/share/git/contrib/completion/git-completion.bash $out/etc/bash_completion.d/
       ln -s $out/share/git/contrib/completion/git-prompt.sh $out/etc/bash_completion.d/
```

Comment:
Shouldn't git-prompt.sh also be linked?

Reply:
The `git-prompt.sh` file contains support functions for setting up a custom prompt. It has nothing to do with bash completion (despite its location).

The old bash completion system eagerly loaded all files in `etc/bash_completion.d`.

I imagine someone took advantage of this eager loading and installed `git-prompt.sh` into `etc/bash_completion.d` to have the functions automatically available without requiring the user to manually having to source the file.

With the new location (`share/bash_completion/completions`) files are loaded on demand, that is, one would have to type `«command»` and press tab, before `«command».sh` gets loaded from `share/bash_completion/completions`.

Therefore it doesn’t make sense to move `git-prompt.sh` to the new location.

It should probably never have been installed in the old location, and with bash completion v1.90 or later, it is not automatically sourced anymore, so if anything should be done, it would probably be to not touch the file at all, but have users source it from `share/git/contrib`.